tools/mo/openvino/tools/mo/front/tf/FifoQueueDequeueCut.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+from collections import defaultdict
+
+from openvino.tools.mo.front.common.replacement import FrontReplacementPattern
+from openvino.tools.mo.front.extractor import add_input_ops
+from openvino.tools.mo.front.output_cut import OutputCut
+from openvino.tools.mo.front.user_data_repack import UserDataRepack
+from openvino.tools.mo.graph.graph import Graph
+from openvino.tools.mo.middle.passes.convert_data_type import np_data_type_to_precision, SUPPORTED_DATA_TYPES
+from openvino.tools.mo.utils.error import Error
+
+
+class FifoQueueDequeueCut(FrontReplacementPattern):
+    """
+    Cuts FifoQueue -> QueueDequeue pattern
+    in order to enable Out Of the Box (OOB) usage.
+    Pass runs only if user didn't specify any input names and shapes.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['cmd_params'].input is None]
+
+    def run_before(self):
+        return [OutputCut]
+
+    def run_after(self):
+        return [UserDataRepack]
+
+    def find_and_replace_pattern(self, graph: Graph):
+        fifo_qd_shapes = defaultdict(list)
+        for node in graph.get_op_nodes():
+            if node.op not in ["QueueDequeue", "QueueDequeueV2"]:
+                continue
+
+            fifo_qd_name = node.soft_get('name', node.id)
+            for port in node.out_nodes().keys():
+                if not np_data_type_to_precision(node.types[port]) in SUPPORTED_DATA_TYPES:
+                    raise Error("Data type {} is not supported for the node {}".format(node.types[port], fifo_qd_name))
+
+                fifo_qd_shapes[fifo_qd_name].append(dict(
+                    shape=node.shapes[port],
+                    out=port,
+                    data_type=node.types[port]
+                ))
+
```

Comment:
please add warning about new inputs
log.warning('Found TF {} operation in the model. '
                    'PLEASE NOTE, the model will contain new inputs {}:0-{} created due to automatically triggered pruning'
                    'transformation for this operation'.format(node.op, node_name, num_outputs))